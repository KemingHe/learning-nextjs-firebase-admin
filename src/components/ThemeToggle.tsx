export default function ThemeToggle(): JSX.Element {
  return (
    <label className="flex cursor-pointer gap-2" htmlFor="themeToggle">
      <span className="label-text">Toggle Theme</span>
      <input
        aria-label="Toggle theme"
        className="toggle theme-controller"
        id="themeToggle"
        type="checkbox"
        value="light"
      />
    </label>
  );
}
