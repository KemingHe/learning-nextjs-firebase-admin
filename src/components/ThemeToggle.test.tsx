import { type RenderResult, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import ThemeToggle from "./ThemeToggle";

describe("ThemeToggle component", () => {
  it("renders the toggle theme label and checkbox", () => {
    render(<ThemeToggle />);

    // Select the label by its text.
    const labelElement = screen.getByText("Toggle Theme");
    expect(labelElement).toBeVisible();

    // Select the checkbox by its role.
    const checkboxElement = screen.getByRole("checkbox", {
      name: "Toggle theme"
    });
    expect(checkboxElement).toBeVisible();

    // Skipping the assertion for the checkbox value
    // because it is not a controlled component.
  });

  it("matches the snapshot", () => {
    const { asFragment }: RenderResult = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
