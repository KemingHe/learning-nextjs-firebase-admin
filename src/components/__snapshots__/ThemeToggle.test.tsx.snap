// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeToggle component > matches the snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex cursor-pointer gap-2"
    for="themeToggle"
  >
    <span
      class="label-text"
    >
      Toggle Theme
    </span>
    <input
      aria-label="Toggle theme"
      class="toggle theme-controller"
      id="themeToggle"
      type="checkbox"
      value="light"
    />
  </label>
</DocumentFragment>
`;
